packages/analytics-js/src/components/userSessionManager/UserSessionManager.ts

```diff
@@ -58,70 +78,119 @@ class UserSessionManager implements IUserSessionManager {
    * Initialize User session with values from storage
    * @param store Selected store
    */
-  init(store: IStore | undefined) {
-    if (!store) {
-      this.setDefaultValues();
-    } else {
-      this.store = store;
+  init(storeManager: IStoreManager) {
+    const sessionTrackingStorage: StorageType = state.storage.entries.value.sessionInfo.storage;
+    this.storeManager = storeManager;
+    this.logger?.debug(state.storage.entries.value);
 
+    if (state.storage.migrate.value === true) {
       this.migrateStorageIfNeeded();
+    } else {
+      this.migrateDataFromPreviousStorage();
+    }
```

Comment:
The migration of previous storages is independent of the `migrate` flag. In fact, migration from previous storages should happen first.

Reply:
> In fact, migration from previous storages should happen first.

No encryption migration is needed first otherwise migration from the previous storage method won't be able to fetch the data.

Reply:
Correct. However, you need to decouple storage migration from the encrypted data `migrate` flag.

Reply:
Done